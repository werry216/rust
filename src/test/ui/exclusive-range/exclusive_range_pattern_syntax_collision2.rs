#![feature(exclusive_range_pattern)]

fn main() {
    match [5..4, 99..105, 43..44] {
        [_, 99..] => {},
        //~^ ERROR `X..` range patterns are not supported
        //~| ERROR arbitrary expressions aren't allowed in patterns
        //~| ERROR pattern requires 2 elements but array has 3
        //~| ERROR only char and numeric types are allowed in range patterns
        _ => {},
    }
}
