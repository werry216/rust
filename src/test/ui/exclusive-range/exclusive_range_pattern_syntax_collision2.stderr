error: `X..` range patterns are not supported
  --> $DIR/exclusive_range_pattern_syntax_collision2.rs:5:13
   |
LL |         [_, 99..] => {},
   |             ^^^^ help: try using the maximum value for the type: `99..MAX`

error: arbitrary expressions aren't allowed in patterns
  --> $DIR/exclusive_range_pattern_syntax_collision2.rs:5:13
   |
LL |         [_, 99..] => {},
   |             ^^^^

error[E0527]: pattern requires 2 elements but array has 3
  --> $DIR/exclusive_range_pattern_syntax_collision2.rs:5:9
   |
LL |         [_, 99..] => {},
   |         ^^^^^^^^^ expected 3 elements

error[E0029]: only char and numeric types are allowed in range patterns
  --> $DIR/exclusive_range_pattern_syntax_collision2.rs:5:13
   |
LL |         [_, 99..] => {},
   |             ^^^^ ranges require char or numeric types
   |
   = note: start type: {integer}
   = note: end type: [type error]

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0029, E0527.
For more information about an error, try `rustc --explain E0029`.
